metricbeat/module/elasticsearch/node_stats/data.go

```diff
@@ -98,13 +160,84 @@ var (
 			"ms":    c.Int("collection_time_in_millis"),
 		},
 	}
+
+	threadPoolSchema = s.Schema{
+		"threads":   c.Int("threads"),
+		"queue":     c.Int("queue"),
+		"active":    c.Int("active"),
+		"rejected":  c.Int("rejected"),
+		"largest":   c.Int("largest"),
+		"completed": c.Int("completed"),
+	}
+
+	ingestPipelineSchema = s.Schema{
+		"count":   c.Int("count"),
+		"current": c.Int("current"),
+		"failed":  c.Int("failed"),
+		"ms":      c.Int("time_in_millis"),
+	}
+
+	adaptiveSelectionSchema = s.Schema{
+		"outgoing_searches": c.Int("outgoing_searches"),
+		"queue_size": s.Object{
+			"avg": c.Int("avg_queue_size"),
+		},
+		"service_time": s.Object{
+			"avg": s.Object{
+				"ns": c.Int("avg_service_time_ns"),
+			},
+		},
+		"response_time": s.Object{
+			"avg": s.Object{
+				"ns": c.Int("avg_response_time_ns"),
+			},
+		},
+		"rank": c.Str("rank"),
+	}
 )
 
 type nodesStruct struct {
 	ClusterName string                            `json:"cluster_name"`
 	Nodes       map[string]map[string]interface{} `json:"nodes"`
 }
 
+func pipelineMetricsMapping(node map[string]interface{}, metricSetFields common.MapStr, r mb.ReporterV2) {
+	pipelines, ok := node["ingest"].(map[string]interface{})["pipelines"].(map[string]interface{})
```

Comment:
@ruflin Is this multi-level indexing into the map + type assertion, all in one step, safe to do or should I break this up into separate steps with error checking after each step?